meson.build

```diff
@@ -0,0 +1,216 @@
+
+foreach script: ['tools/ds-identify', 'tools/hook-hotplug','tools/uncloud-init', 'tools/write-ssh-key-fingerprints' ]
+  install_data(
+    script, install_dir: lib_exec_dir,  install_tag: 'scripts')
+endforeach
+
+
+# Scripts
+foreach script: files(
+  ['tools/cloud-init-per', 'tools/cloud-id', 'tools/cloud-init'])
+  install_data(
+    script,
+    install_mode: 'rwxr-xr-x',
+    install_dir: get_option('bindir'),
+  )
+endforeach
+
+# Schemas
+
+# Required Config and Templates
+render_tmpl = './tools/render-template'
+subdir('config/cloud.cfg.d')
+subdir('templates')
+
+if INIT_SYSTEM == 'systemd'
+  udev = dependency('udev')
+  systemd_unit_dir =  systemd.get_variable(
+    pkgconfig: 'systemdsystemunitdir'
+  )
+  systemd_generator_dir =  systemd.get_variable(
+    pkgconfig: 'systemdsystemgeneratordir'
+  )
+  subdir('systemd')
+  udev_dir = udev.get_variable(
+    pkgconfig: 'udevdir', default_value: '/usr/lib/udev'
+  )
+  install_data(
+    'udev/66-azure-ephemeral.rules',
+    install_dir: join_paths(udev_dir, 'rules.d'),
+    install_tag: 'systemd'
+  )
+
+  # Must generate systemd templates in root mesonbuild, because nested
+  # systemd/meson.build results in builddir @OUTPUT@ macro being doubly-nested
+  # paths
+  systemd_generators = run_command(find, 'systemd', '-name', '*generator*.tmpl', check: true).stdout().strip().split('\n')
+  foreach template: systemd_generators
+    custom_target(
+      input: template,
+      output: '@BASENAME@',
+      command: [render_tmpl, '@INPUT@', join_paths(meson.current_build_dir(), '@OUTPUT@')],
+      install: true,
+      install_dir: systemd_generator_dir,
+      install_tag: 'systemd'
+    )
+  endforeach
+
+  systemd_templates = run_command(find, 'systemd', '-not', '-name', '*generator*', '-name', '*.tmpl', check: true).stdout().strip().split('\n')
+  foreach template: systemd_templates
+    custom_target(
+      input: template,
+      output: '@BASENAME@',
+      command: [render_tmpl, '@INPUT@', join_paths(meson.current_build_dir(), '@OUTPUT@')],
+      install: true,
+      install_dir: systemd_unit_dir,
+      install_tag: 'systemd'
+    )
+  endforeach
+endif
+
+# Schema files
+json_schemas = run_command(find, 'cloudinit', '-name', '*.json', check: true).stdout().strip().split('\n')
+python.install_sources(json_schemas, subdir: join_paths('cloudinit', 'config', 'schemas'))
+
+custom_target(
+  input: 'config/cloud.cfg.tmpl',
+  output: 'cloud.cfg',
+  command: [render_tmpl, '--is-yaml', '@INPUT@', join_paths(meson.current_build_dir(), '@OUTPUT@')],
+  install: true,
+  install_dir: join_paths(sysconfdir, 'cloud'),
+  install_tag: 'config'
+)
+
+
+# Docs and Examples:
+subdir('doc/examples')
+subdir('doc/examples/seed')
+mandir = get_option('mandir')
+foreach man_page : ['cloud-init.1', 'cloud-id.1', 'cloud-init-per.1']
+  install_data(
+    'doc/man/' + man_page,
+    install_dir: join_paths(mandir,  'man1'),
+    install_tag: 'docs'
+  )
+endforeach
```

Comment:
[install_man()](https://mesonbuild.com/Reference-manual_functions.html#install_man) looks like it would eliminate the need for the mandir lookup.